pkg/validation/schema.go

```diff
@@ -0,0 +1,86 @@
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/iawaknahc/jsonschema/pkg/jsonpointer"
+	"github.com/iawaknahc/jsonschema/pkg/jsonschema"
+	"io"
+	"strings"
+)
+
+type SimpleSchema struct {
+	col *jsonschema.Collection
+}
+
+func NewSimpleSchema(schema string) *SimpleSchema {
+	col := jsonschema.NewCollection()
+	col.AddSchema(strings.NewReader(schema), "")
+	return &SimpleSchema{
+		col: col,
+	}
+}
+
+func (s *SimpleSchema) ValidateReader(r io.Reader) error {
+	return convertErrors(validateSchema(s.col, r))
+}
+
+type MultipartSchema struct {
+	mainPartID string
+	parts      map[string]interface{}
+	col        *jsonschema.Collection
```

Comment:
We are not utilizing `$anchor` and the first argument of `Apply`. So we have to build a MultipartSchema for each schema we want to validate against, which may become tedious.